### PR TITLE
Clean up clippy warnings

### PR DESCRIPTION
**Pull request by majorz**
Connects-To: #68

### CODE REVIEW

File: src/connection.rs [hunk 1]
```diff
@@ -180,7 +180,6 @@ pub enum ConnectionState {
 impl From<i64> for ConnectionState {
     fn from(state: i64) -> Self {
         match state {
-            0 => ConnectionState::Unknown,
```
Comment (josephroberts) [inline src/connection.rs]: Why did you remove this? It' a valid state https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NMState

Comment (majorz) [inline src/connection.rs]: Clippy complained about the duplication with `_ => ConnectionState::Unknown,`, which is situated at the end of the match statement.

Comment (josephroberts) [inline src/connection.rs]: I think we should keep `0` as its a valid state and change the default `_` state to throw some kind of error as it should never be hit.

Comment (majorz) [inline src/connection.rs]: Maybe just a `warn!(...)` log entry? Otherwise it will complicate the code too much and I doubt we will even encounter this scenario - unless NM API adds new states in the future, which we can treat as unknown until we add them.

Comment (josephroberts) [inline src/connection.rs]: sounds good, it may be worth adding the extra states to the match statement aswell
